app/models/mapping_term.rb

```diff
@@ -14,4 +14,21 @@ class MappingTerm < ApplicationRecord
   has_and_belongs_to_many :mapped_terms, join_table: :mapping_term_mapped_terms, class_name: :Term
 
   validates :uri, presence: true
+
+  ###
+  # @description: Give the params the correct format. This is in case we receive an array of
+  #   mapped_terms. We need ot be instantiate each term in order to persist
+  # @param [Array] mapped_term_ids: A collection of ids representing the terms that are
+  #   being mapped in this alignment
+  ###
+  def update_mapped_terms mapped_term_ids
+    # Instantiate each mapped term
+    mapped_terms_collection =
+      (mapped_term_ids - mapped_terms.pluck(:id)).map do |mapped_term_id|
+        Term.find(mapped_term_id)
```

Comment:
What's the point of instantiating the terms here? Can't we add them by the IDs:

```ruby
  mapped_term_ids += (mapped_terms.pluck(:id) - mapped_term_ids)
```

Reply:
@excelsior there's no such attribute `mapped_ter_ids` its the name of the param. We're modeling a "many to many" relationship here, so a valid way to perform the update is by instantiating the Term (or collection of Terms) and then adding to the relationship name (`mapped_terms` in this case).

The other way is by adding a class for the join table, which in this case I consider unnecessary due to the fact that we don't have any additional information to store on such model.

Reply:
Sorry, my comment was somewhat confusing. 

Adding the `has_and_belongs_to_many :mapped_terms` association generates the `mapped_term_ids` and `mapped_term_ids=` methods (among others):
https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_and_belongs_to_many

Hence, this method can be simplified to something like this:

```ruby
def update_mapped_terms(term_ids)
  self.mapped_term_ids += (term_ids - mapped_term_ids)
end
```